modules/tinymce/src/themes/silver/test/ts/browser/editor/core/LineHeightTest.ts

```diff
@@ -0,0 +1,155 @@
+    Assertions.assertEq('Toolbar contains the correct line heights', ideal, actual);
+  })
+]);
+
+UnitTest.asyncTest('LineHeightTest', (success, failure) => {
+  Theme();
+
+  const settings = {
+    base_url: '/project/tinymce/js/tinymce',
+    toolbar: 'line-height'
+  };
+
+  Chain.pipeline([
+    McEditor.cFromSettings(settings),
+    Chain.runStepsOnValue((editor: Editor) => {
+      const api = TinyApis(editor);
+      const ui = TinyUi(editor);
+
+      return [
+        Log.stepsAsStep('TINY-4843', 'Toolbar lists correct line heights', [
+          sOpenToolbar(ui),
+          sAssertOptions(ui, 'toolbar', [ '1', '1.1', '1.2', '1.3', '1.4', '1.5', '2' ], Optional.some('1.4')),
+          sCloseToolbar(ui)
+        ]),
+
+        Log.stepsAsStep('TINY-4843', 'Menu lists correct line heights', [
+          sOpenMenu(ui),
+          sAssertOptions(ui, 'menu', [ '1', '1.1', '1.2', '1.3', '1.4', '1.5', '2' ], Optional.some('1.4')),
+          sCloseMenu(ui)
+        ]),
+
+        Log.stepsAsStep('TINY-4843', 'Toolbar can alter line height', [
+          api.sSetContent('<p>Hello</p>'),
+          api.sSetCursor([ 0, 0 ], 0),
+          sOpenToolbar(ui),
+          sSelectLineHeight(ui, 'toolbar', '1.5'),
+          api.sAssertContent('<p style="line-height: 1.5;">Hello</p>')
+        ]),
+
+        Log.stepsAsStep('TINY-4843', 'Menu can alter line height', [
+          api.sSetContent('<p>Hello</p>'),
+          api.sSetCursor([ 0, 0 ], 0),
+          sOpenMenu(ui),
+          sSelectLineHeight(ui, 'menu', '1.5'),
+          api.sAssertContent('<p style="line-height: 1.5;">Hello</p>')
+        ]),
+
+        Log.stepsAsStep('TINY-4843', 'Toolbar only shows values within settings', [
+          api.sSetContent('<p style="line-height: 30px;">Hello</p>'),
+          api.sSetCursor([ 0, 0 ], 0),
+          sOpenToolbar(ui),
+          sAssertOptions(ui, 'toolbar', [ '1', '1.1', '1.2', '1.3', '1.4', '1.5', '2' ], Optional.none()),
+          sCloseToolbar(ui)
+        ]),
+
+        Log.stepsAsStep('TINY-4843', 'Menu shows currently selected line height', [
+          api.sSetContent('<p style="line-height: 30px;">Hello</p>'),
+          api.sSetCursor([ 0, 0 ], 0),
+          sOpenMenu(ui),
+          sAssertOptions(ui, 'menu', [ '1', '1.1', '1.2', '1.3', '1.4', '1.5', '2' ], Optional.none()),
+          sCloseMenu(ui)
+        ]),
+
+        Log.stepsAsStep('TINY-4843', 'Toolbar updates if line height changes', [
+          api.sSetContent('<p>Hello</p>'),
+          api.sSetCursor([ 0, 0 ], 0),
+          sOpenToolbar(ui),
+          sAssertOptions(ui, 'menu', [ '1', '1.1', '1.2', '1.3', '1.4', '1.5', '2'], Optional.some('1.4')),
+          api.sExecCommand('LineHeight', '1.1'),
+          sAssertOptions(ui, 'menu', [ '1', '1.1', '1.2', '1.3', '1.4', '1.5', '2'], Optional.some('1.1')),
+          sCloseToolbar(ui)
+        ]),
+
+        Log.stepsAsStep('TINY-4843', 'Toolbar updates if line height changes', [
```

Comment:
I think the label has to be updated to be `menu` here.

Reply:
Oh yeah good point, this file is a mess for that. Should be good now